tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,266 @@
+                if pkt["LACPRetryCount"].version == 0xf1:
+                    if pkt["LACPRetryCount"].actor_system == actorInfo["system"] \
+                            and pkt["LACPRetryCount"].actor_port_number == actorInfo["port"] \
+                            and pkt["LACPRetryCount"].partner_system == partnerInfo["system"] \
+                            and pkt["LACPRetryCount"].partner_port_number == partnerInfo["port"]:
+                        sendVersionVerified = True
+                    elif pkt["LACPRetryCount"].actor_system == partnerInfo["system"] \
+                            and pkt["LACPRetryCount"].actor_port_number == partnerInfo["port"] \
+                            and pkt["LACPRetryCount"].partner_system == actorInfo["system"] \
+                            and pkt["LACPRetryCount"].partner_port_number == actorInfo["port"]:
+                        receiveVersionVerified = True
+            pytest_assert(sendVersionVerified, "unable to verify that LACPDU packets sent were the right version")
+            pytest_assert(receiveVersionVerified,
+                          "unable to verify that LACPDU packets received were the right version")
+
+
+class TestNeighborRetryCount:
+    def test_peer_retry_count(self, duthost, nbrhosts, higher_retry_count_on_peers):
+        """
+        Test that DUT sees new retry count when peers update retry count.
+        """
+        for nbr in list(nbrhosts.keys()):
+            port_channel_status = nbrhosts[nbr]['host'].get_port_channel_status("PortChannel1")
+            pytest_assert(port_channel_status["runner"]["retry_count"] == 5, "retry count on neighbor is incorrect")
+
+        cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+        port_channels = cfg_facts["PORTCHANNEL"].keys()
+        for port_channel in port_channels:
+            port_channel_status = duthost.get_port_channel_status(port_channel)
+            number_of_lag_member = len(cfg_facts["PORTCHANNEL_MEMBER"][port_channel])
+            pytest_assert("ports" in port_channel_status and number_of_lag_member == len(port_channel_status["ports"]),
+                          "get port status error")
+            for _, status in list(port_channel_status["ports"].items()):
+                pytest_assert(status["runner"]["selected"], "status of lag member error")
+                pytest_assert(status["runner"]["partner_retry_count"] == 5, "partner retry count is incorrect")
+
+    def test_peer_retry_count_packet_version(self, duthost, nbrhosts, higher_retry_count_on_peers, scapy_lacp_layer):
+        """
+        Test that peers and DUT use new LACPDU version when peers use a non-standard retry count
+        """
+        check_lacpdu_packet_version(duthost)
+
+    def test_kill_teamd_lag_up(self, duthost, nbrhosts, higher_retry_count_on_peers, config_reload_on_cleanup):
+        """
+        Test that the lag remains up for 150 seconds after killing teamd on the peer
```

Comment:
Is this a valid expectation? If T1 side things go bad, we do not want T0 to still hold routes from that T1. IF LAGs stay up for 150s, BGP may still be up and routes still not withdrawn.

Reply:
I think that's the cost of keeping the LAG up for 150 seconds; it's a balance between detecting and reacting to failures quickly, and bringing down the link for minor issues (or in our case, bringing down the link too soon during warm upgrade).

Reply:
I still think there is room for improvement here.

Since the functionality is primarily for warm-reboot context, then should the LAGs teardown early (even w/ high retry count) if the warm-reboot system wide flag is NOT set?
Only allow delayed teardown if the system is undergoing warm-reboot. I understand that this is not a test related question, rather this is the feature design specific.